service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DraftInvoiceGenerator.kt

```diff
@@ -161,6 +180,14 @@ class DraftInvoiceGenerator(
             .groupBy { (_, stub) -> stub.child }
             .flatMap { (child, childStubs) ->
                 val separatePeriods = mergePeriods(childStubs)
+
+                val logic = invoiceGenerationLogicChooser.logicForMonth(
+                    separatePeriods.first().first.start.year,
+                    separatePeriods.first().first.start.month,
```

Comment:
Why not just use `invoicePeriod` instead of `separatePeriods.first().first`?

Reply:
Fixed!